mutiny-core/src/node.rs

```diff
@@ -1617,6 +1617,7 @@ impl<S: MutinyStorage> Node<S> {
         message: Option<String>,
         labels: Vec<String>,
         payment_id: PaymentId,
+        preimage: Option<[u8; 32]>,
```

Comment:
this is used from the NWC stuff, allows for specifying a preimage for the keysend.

We need it for ourself as well so we don't generate a completely random preimage for the event, need to do it deterministically so if we process the event multiple times we won't send the payment multiple times

Reply:
I'm generating the preimage deterministically from the nwc request event in the `handle_nwc_keysend_payment` method and passing it as an option. Got from @benthecarman that this would be needed to avoid making another keysend payment for the same event in case we try to process that same request event more than once.

Reply:
whoops, saw your comment after posted mine lol

Reply:
I looked and didn't see it. Where is it?

Reply:
setting that optional preimage here https://github.com/elnosh/mutiny-node/blob/nwc-commands/mutiny-core/src/nostr/nwc.rs#L1258-L1269

and passing it here https://github.com/elnosh/mutiny-node/blob/nwc-commands/mutiny-core/src/nostr/nwc.rs#L1332-L1334